bundles/org.openhab.binding.semsportal/src/main/java/org/openhab/binding/semsportal/internal/SEMSPortalHandler.java

```diff
@@ -0,0 +1,268 @@
+        gson = new GsonBuilder().setDateFormat(SEMSPortalBindingConstants.DATE_FORMAT).create();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (!loggedIn) {
+            logger.debug("Not logged in. Ignoring command {} for channel {}", command, channelUID);
+            return;
+        }
+        if (command instanceof RefreshType) {
+            scheduler.execute(() -> {
+                updateChannelState(channelUID);
+            });
+        }
+    }
+
+    private void updateChannelState(ChannelUID channelUID) {
+        if (!loggedIn) {
+            logger.debug("Not logged in. Ignoring update request for channel {}", channelUID.getAsString());
+            return;
+        }
+        ensureRecentStatus();
+        if (CHANNEL_CURRENT_OUTPUT.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getCurrentOutput(currentStatus));
+        } else if (CHANNEL_TODAY_TOTAL.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getDayTotal(currentStatus));
+        } else if (CHANNEL_MONTH_TOTAL.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getMonthTotal(currentStatus));
+        } else if (CHANNEL_OVERALL_TOTAL.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getOverallTotal(currentStatus));
+        } else if (CHANNEL_ONLINE_STATE.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getOnline(currentStatus));
+            if (currentStatus != null && currentStatus.isOnline() != stationOnline) {
+                stationOnline = currentStatus.isOnline();
+                triggerChannel(channelUID, OnOffType.from(stationOnline).name());
+            }
+        } else if (CHANNEL_TODAY_INCOME.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getDayIncome(currentStatus));
+        } else if (CHANNEL_TOTAL_INCOME.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getTotalIncome(currentStatus));
+        } else if (CHANNEL_LASTUPDATE.equals(channelUID.getId())) {
+            updateState(channelUID.getId(), StateHelper.getLastUpdate(currentStatus));
+        }
+    }
+
+    private void ensureRecentStatus() {
+        if (lastUpdate.isBefore(LocalDateTime.now().minusMinutes(MAX_STATUS_AGE_MINUTES))) {
+            updateStation();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SEMSPortalConfiguration.class);
+        if (!config.isProperlyInitialized()) {
+            logger.warn("Username, password or station missing. Please provide the right values in the configuration.");
+            updateStatus(ThingStatus.UNINITIALIZED, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Check username / password / station settings");
```

Comment:
See the guidelines for logging. Remove the logger and just use the ThingStatus here, no need to do both.

Reply:
done